src/martialarts.cpp

```diff
@@ -74,6 +68,46 @@ void add_if_exists( JsonObject &jo, Container &cont, bool was_loaded,
     }
 }
 
+class ma_skill_reader : public generic_typed_reader<ma_skill_reader>
+{
+    public:
+        std::pair<skill_id, int> get_next( JsonIn &jin ) const {
+            JsonObject jo = jin.get_object();
+            return std::pair<skill_id, int>( skill_id( jo.get_string( "name" ) ), jo.get_int( "level" ) );
+        }
+        template<typename C>
+        void erase_next( JsonIn &jin, C &container ) const {
+            const skill_id id = skill_id( jin.get_string() );
+            reader_detail::handler<C>().erase_if( container, [&id]( const std::pair<skill_id, int> &e ) {
+                return e.first == id;
+            } );
+        }
+};
+
+class ma_weapon_damage_reader : public generic_typed_reader<ma_weapon_damage_reader>
+{
+    public:
+        std::map<std::string, damage_type> dt_map = get_dt_map();
+
+        std::pair<damage_type, int> get_next( JsonIn &jin ) const {
+            JsonObject jo = jin.get_object();
+            std::string type = jo.get_string( "type" );
+            std::string s = dt_map.find( type ) != dt_map.end() ? type : "true";
+            const damage_type dt = dt_map.find( s )->second;
+            return std::pair<damage_type, int>( dt, jo.get_int( "min" ) );
+        }
+        template<typename C>
+        void erase_next( JsonIn &jin, C &container ) const {
+            JsonObject jo = jin.get_object();
+            std::string type = jo.get_string( "type" );
+            std::string s = dt_map.find( type ) != dt_map.end() ? type : "true";
+            damage_type id = dt_map.find( s )->second;
```

Comment:
Suggestion (only works when `get_dt_map` returns a reference):
```C++
const auto iter = get_dt_map().find( s );
if( iter == get_dt_map().end() ) {
    jin.throw_error("Invalid damage type");
}
const damage_type id = iter->second;
```
This avoids the repeated lookup in the map (the two calls to `find`).